### ISSUE TITLE
Implement Weighted Loss for Class Imbalance

### ISSUE THREAD

**Issue by zebulonmcknight** [zebulonmcknight/cs325_final#7]
Description:
Address potential class imbalance across the 120 dog breeds by integrating a weighted cross-entropy loss function. Each class should be assigned a weight inversely proportional to its frequency in the dataset.

Tasks:

Compute class frequencies from the training set.

Derive normalized weights (e.g., using 1 / frequency or 1 / sqrt(frequency)).

Modify the loss function:

criterion = nn.CrossEntropyLoss(weight=class_weights)

Retrain the model using both weighted and unweighted loss, and compare results.

Visualize the effect on underrepresented classes if possible (per-class accuracy).

Deliverables:

Updated model training script with weighted loss.

Plots comparing validation accuracy and loss before/after weighting.

Short write-up summarizing whether weighting improved balance.
